src/server/auth/server/api_server.go

```diff
@@ -698,11 +741,6 @@ func (a *apiServer) ModifyRoleBindingInTransaction(
 		return nil, err
 	}
 
-	roles, err := rolesFromRoleSlice(req.Roles)
```

Comment:
I refactored the non-auth guts of `ModifyRoleBindingInTransaction` into `setUserRoleBindingInTransaction` so it can be reused in each of the pipeline methods.